pages/_app.js

```diff
@@ -79,7 +96,24 @@ export default function App({ Component, pageProps }) {
     } else {
       setPairingsInfo([...pairingsInfo, { _id, isFavorite: true }]);
     }
-  }
+  };
+
+  const updatePairingRating = (_id, newRating) => {
+    const updatedPairings = pairings.map((pairing) =>
+      pairing._id === _id
+        ? {
+            ...pairing,
+            totalRatings: pairing.totalRatings + 1,
+            rating:
+              (pairing.rating * pairing.totalRatings + newRating) /
+              (pairing.totalRatings + 1),
+          }
+        : pairing
+    );
+
+    setPairings(updatedPairings);
+    localStorage.setItem("pairings", JSON.stringify(updatedPairings));
```

Comment:
You are setting things to the localStorage twice here. Once with the useLocalStorageState hook and once manually.

```suggestion
    setPairings(updatedPairings);
```